### PR TITLE
My homework

### PR DESCRIPTION
**Pull request by Pranav-PSU**
Homework 03 pull request

### CODE REVIEW

File: hw3/01-routing.js [hunk 1]
```diff
@@ -1,51 +1,93 @@
...
+        res.write(present);
+        res.end();
+    } else {
+        res.writeHead(404, {
+            'Content-Type': 'text/html',
+        });
+        res.write('404 - page not found');
+        res.end();
+    }
 });
```
Comment (rutujapadgilwar) [inline hw3/01-routing.js:103]: Everything looks good but just a small coding recommendation would be using a switch-case.

File: hw3/02-url.js [hunk 2]
```diff
@@ -3,40 +3,48 @@ const http = require('http');
...
+        res.write(`<ul> ${routeResults} </ul>`);
+    } else if (req.url === '/attributes/') {
+        console.log(req.query);
+    }
+
+    const queryParameters = url.searchParams;
+    let cssStyle = '"border:1px solid black"';
+    let tableHTML = `<table style = ${cssStyle}>`;
+    queryParameters.forEach((element, index) => {
+        tableHTML += `<tr style=${cssStyle}><td style=${cssStyle}>${index}</td><td style=${cssStyle}>${element}</td>`;
```
Comment (rutujapadgilwar) [inline hw3/02-url.js:37]: Everything looks totally fine and no comments for this one.

File: hw3/03-form.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+var bodyParser = require('body-parser');
+const app = express();
+
+const port = process.env.PORT || 5001;
+
+// http://localhost:5001/form should return a form with input elements for username, email, and submit button
+
+// http://localhost:5001/submit should return all the data the user entered
+
+app.get('/', (req, res) => {
```
Comment (rutujapadgilwar) [inline hw3/03-form.js]: I believe app.get is a functionality in Express JS. Can you check and confirm if we are allowed to use it in this assignment as I have submitted my code using Node JS only.

**Review by rutujapadgilwar — COMMENTED** [review]
Added comments for HW3 specific files

**Comment by AdrikGG** [comment]
Your code looks pretty good! I like the use of switch case, good idea to implement that suggestion. I'm pretty sure you are allowed to use Express.js for Exercise 03 since the instruction file states you must use just Node.js only for exercises 01 and 02, not 03, but I could be wrong. 

There are a couple things I noticed in your 03 implementation that you may want to change. On line 18 you use `==`, instead use `===`, on line 19 you use `==` again, but here you don't have to check equality to `true`, you can just check `Value`. Ex: `Value ? statement_if_true : statement_if_false`. I also noticed you were passing the form data in the URL with a GET requests. This works fine, but you generally want to pass data like this (that could contain sensitive information) in the body with a POST request.

One last thing, in your 03-form.html, most of your `label` elements are not bound to `input` elements. (`for="searchterm"` on lines 18, 20, and 22).

**Comment by Pranav-PSU** [comment]
PR for homework 3 is Merged

**Pull request closed without merging**